src/opnsense/mvc/app/controllers/OPNsense/DHCPv6/Api/LeasesController.php

```diff
@@ -0,0 +1,245 @@
+                if (!empty((string)$if->ipaddrv6) && Util::isIpAddress((string)$if->ipaddrv6)) {
+                    $intf = (string)$if->descr;
+                    $leases[$idx]['if_descr'] = $intf;
+                    $leases[$idx]['if'] = $lease['if'];
+
+                    if (!array_key_exists($lease['if'], $interfaces)) {
+                        $interfaces[$lease['if']] = $intf;
+                    }
+                }
+            } else {
+                foreach ($config->dhcpdv6->children() as $dhcpif => $dhcpifconf) {
+                    $if = $config->interfaces->$dhcpif;
+                    if (!empty((string)$if->ipaddrv6) && Util::isIpAddress((string)$if->ipaddrv6)) {
+                        if (Util::isIPInCIDR($lease['address'], (string)$if->ipaddrv6 . '/' . (string)$if->subnetv6)) {
+                            $intf = (string)$if->descr;
+                            $leases[$idx]['if_descr'] = $intf;
+                            $leases[$idx]['if'] = $dhcpif;
+
+                            if (!array_key_exists($dhcpif, $interfaces)) {
+                                $interfaces[$dhcpif] = $intf;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        $response = $this->searchRecordsetBase($leases, null, 'address', function ($key) use ($selected_interfaces) {
+            if (empty($selected_interfaces) || in_array($key['if'], $selected_interfaces)) {
+                return true;
+            }
+
+            return false;
+        }, SORT_REGULAR);
+
+        $response['interfaces'] = $interfaces;
+        return $response;
+    }
+
+    public function searchPrefixAction()
+    {
+        $this->sessionClose();
+        $backend = new Backend();
+        $prefixes = [];
+
+        $raw_leases = json_decode($backend->configdpRun('dhcpd list leases6 1'), true);
+        foreach ($raw_leases as $raw_lease) {
+            if ($raw_lease['lease_type'] === 'ia-pd' && array_key_exists('prefixes', $raw_lease)) {
+                $prefix = [];
+                $prefix['lease_type'] = $raw_lease['lease_type'];
+                $prefix['iaid'] = $raw_lease['iaid'];
+                $prefix['duid'] = $raw_lease['duid'];
+                if (array_key_exists('cltt', $raw_lease)) {
+                    $prefix['cltt'] = date('Y/m/d H:i:s', $raw_lease['cltt']);
+                }
+
+                $prefix_raw = $raw_lease['prefixes'][0];
+                $prefix['prefix'] = $prefix_raw['iaprefix'];
+                if (array_key_exists('ends', $prefix_raw)) {
+                    $prefix['ends'] =  date('Y/m/d H:i:s', $prefix_raw['ends']);
+                }
+                $prefix['state'] = $prefix_raw['binding'] == 'free' ? 'expired' : $prefix_raw['binding'];
+                $prefixes[] = $prefix;
+            }
+        }
+
+        $response = $this->searchRecordsetBase($prefixes, null, 'prefix', null, SORT_REGULAR);
```

Comment:
`return $this->searchRecordsetBase()` ? storing the value likely doesn't make much sense here.